### ISSUE TITLE
Freezing on Shutdown in Intel mode

### ISSUE THREAD

**Issue by Nyuten** [stockmind/dell-xps-9560-ubuntu-respin#5]
Hi,

I've followed every single step in your instructions and built the ISO with the 18.04 LTS release and bionicbeaver compability.
Everything was fine until I switched the GPU to Intel.
After rebooting my computer freezes on Shutdown leaving me with holding the power button being the only option...
Did you encounter and maybe fix this issue?

**Comment by stockmind** [comment]
I have to ask to the friend of mine that owns the laptop: @zangobot

Can you elaborate more about the computer freeze?
Can you post a photo of what you see on screen when it freeze?
It freezes right after you press the shoutdown button?
Or it freezes  on the "shoutdown log" screen after the display server gets closed and dispatched?

**Comment by Nyuten** [comment]
Yes, it freezes right after pressing the shutdown button, or entering `sudo shutdown now`.
After trying to shutdown, it leaves me with the screen without the shutdown menu overlay and cursor.
Mouse and keyboard are not responding, and i can't switch to TTY.

![img_20180428_224201](https://user-images.githubusercontent.com/5490411/39400739-85fe2b44-4b36-11e8-8884-f0d99723980c.jpg)

**Comment by stockmind** [comment]
What kernel version are you using now? What kernel version have you chosen in the respin?
Post output of:
```
uname -r
dmidecode | grep "Release"
dmesg | grep "EFI v"
```

This happens on 17.10? 
Can you try to update to latest mainline kernel? (use something like UKUU)
Do you have the latest bios installed?

Thanks!

**Comment by stockmind** [comment]
@ahmadsocrat You are on 18.04, right? No issue on 17.10?

**Comment by ahmadsocrat** [comment]
Yes, 17.10 had no issue with shutdown.

**Comment by stockmind** [comment]
@ahmadsocrat @Nyuten can you try to update to latest 4.16.7 kernel?

I've also tried to respin an iso with that already packed in with other packages that may help:
https://downloads.cre.ovh/

A new nvidia driver has been released yesterday (396) and maybe could help.

**Comment by ahmadsocrat** [comment]
It improved a bit after upgrading to 4.16.7, but still take a very long time to shutdown, like 1-2 minutes.

**Comment by JackHack96** [comment]
Well, at least now it doesn't hang during shutdown. If it takes 1-2 minutes it's probably due to a systemd unit hang.
At boot, try pressing shift (or esc) for viewing the GRUB menu and press e. Go the line where it says "quiet splash" (it should be the second-last line) and delete "quiet splash", then press F10.

Now you should see a sort of log during startup. When you'll shutdown, the same log will appear, showing you some useful information (like hanging systemd unit)

**Comment by ahmadsocrat** [comment]
Hi,
Thanks a lot for your prompt responses.
I tested as per your instructions, but the UI still hangs without showing anything, then after about 1-2 minutes it shows all log normally and shuts down.

**Comment by JackHack96** [comment]
What if you make the same process, but instead of powering off from gnome you use a TTY (Ctrl+Alt+F3, login name and password and then type poweroff)

**Comment by ahmadsocrat** [comment]
I received the following error message (2 times) about the nouveau driver:

`A stop job is running for Session 2 of user xxxx (25s /1min 30s) [10361.645918] nouveau 0000:01:00.0: systemd-logind[1137]: failed to idle channel 2 (systemd-logind[1137]`

**Comment by JackHack96** [comment]
It seems that you are not using the proprietary drivers, or nouveau is still loaded...

**Comment by ahmadsocrat** [comment]
I am using the iso with the respin script and selected the intel driver
with prime-select, nothing else.

Kind Regards,

Ahmad Amr

**Comment by JackHack96** [comment]
Yes, and I suspect that the nvidia driver is not installed or configured correctly, like in #6

**Comment by ahmadsocrat** [comment]
It works very well when I run prime-select nvidia and then reboot.

Kind Regards,

Ahmad Amr

**Comment by JackHack96** [comment]
Unfortunately I still haven't tested this behavior, I still don't own a Dell XPS (I'll have to wait till June)

**Comment by qhartman** [comment]
I own this machine and will try to help with some testing. It's my work machine though so I'll to be somewhat careful, ideally testing installs to USB or similar to avoid messing with the internal install. I'll drop a note in here if I'm able to replicate.

**Comment by JackHack96** [comment]
I think yes, look here for other interesting details: https://bugs.launchpad.net/ubuntu/+source/nvidia-prime/+bug/1765363

**Comment by ahmadsocrat** [comment]
This solved the issue for me, now my grub line is as follows:

GRUB_CMDLINE_LINUX_DEFAULT="quiet splash acpi_rev_override=1
nouveau.modeset=0"

I am on kernel 4.16.7-041607-generic and nvidia driver 396, power
consumption is around 18 now, which is a major improvement, will keep you
updated with any changes or side effects.

Kind Regards / مع تحياتي

أحمد عمرو
Ahmad Amr

**Comment by ahmadsocrat** [comment]
Hi guys,

With adding powertop auto-tune to startup, now I get around 10 only power
consumption.

I used this article to add powertop auto-tune to startup:
https://askubuntu.com/questions/112705/how-do-i-make-powertop-changes-permanent

I didn't have to run anything else!

Kind Regards / مع تحياتي

أحمد عمرو
Ahmad Amr

**Comment by stockmind** [comment]
Switch to nvidia works as expected after those changes? How are power consumptions on nvidia?

**Comment by Nyuten** [comment]
> Switch to nvidia works as expected after those changes? How are power consumptions on nvidia?
Works as expected and power consumptions are about ~13-19W when idle.

> With adding powertop auto-tune to startup, now I get around 10 only power
consumption.
@ahmadsocrat `sudo powertop --auto-tune` gives me following error message:
`sudo powertop --auto-tune
modprobe cpufreq_stats failedLoaded 491 prior measurements
RAPL device for cpu 0
RAPL Using PowerCap Sysfs : Domain Mask f
RAPL device for cpu 0
RAPL Using PowerCap Sysfs : Domain Mask f
Devfreq not enabled
glob returned GLOB`

Did you encounter this problem?

I managed to cut down consumptions to about 6-12W when idle by completely turning off the nvidia graphic card using bbswitch:
https://github.com/Bumblebee-Project/bbswitch

**Comment by stockmind** [comment]
The @tlvd service posted in issue #8 seems to fix the power issues of Nvidia, @Nyuten this helps you?

**Comment by Nyuten** [comment]
Oh thanks, worked like a charm!

**Comment by CrystalSpore** [comment]
So just to clarify with all the posting, it seems that we have a way to make a stable, power efficient version of 18.04/18.10. Should an official version be compiled and added to this repository if that is the case? (Of course with all the things considered between #5, #6, and #8?)
